bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/discovery/VeluxDiscoveryService.java

```diff
@@ -0,0 +1,236 @@
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VeluxDiscoveryService} is responsible for discovering scenes on
+ * the current Velux Bridge.
+ *
+ * @author Guenther Schreiner - Initial contribution.
+ */
+// ToDo: check whether an immediate activation is preferable.
+// Might be activated by:
+// @Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.velux")
+//
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery.velux")
+public class VeluxDiscoveryService extends AbstractDiscoveryService implements Runnable {
+    private final Logger logger = LoggerFactory.getLogger(VeluxDiscoveryService.class);
+
+    private static final int DISCOVER_TIMEOUT_SECONDS = 300;
+
+    private static @Nullable VeluxBridgeHandler bridgeHandler = null;
+
+    /**
+     * Initializes the {@link VeluxDiscoveryService} without a {@link VeluxBridgeHandler}.
+     * <P>
+     * This is the main entry point for the OSGI DiscoveryService.
+     *
+     */
+    public VeluxDiscoveryService() {
+        super(VeluxBindingConstants.SUPPORTED_THINGS_ITEMS, DISCOVER_TIMEOUT_SECONDS);
+        logger.trace("VeluxDiscoveryService(without Bridge) just initialized.");
+    }
+
+    /**
+     * Initializes the {@link VeluxDiscoveryService} with a reference to the well-prepared environment with a
+     * {@link VeluxBridgeHandler}.
+     *
+     * @param bridge Initialized Velux bridge handler.
+     */
+    public VeluxDiscoveryService(VeluxBridgeHandler bridge) {
+        super(VeluxBindingConstants.SUPPORTED_THINGS_ITEMS, DISCOVER_TIMEOUT_SECONDS);
+        logger.trace("VeluxDiscoveryService(bridge={}) just initialized.", bridge);
+
+        synchronized (this) {
+            if (bridgeHandler == null) {
+                logger.trace("VeluxDiscoveryService(): registering bridge {} for lateron use for Discovery.", bridge);
+            } else if (!bridge.equals(bridgeHandler)) {
+                logger.trace("VeluxDiscoveryService(): replacing already registered bridge {} by {}.", bridgeHandler,
+                        bridge);
+            }
+            bridgeHandler = bridge;
+        }
+    }
+
+    @Override
+    public void deactivate() {
+        logger.trace("deactivate() called.");
+        super.deactivate();
+    }
+
+    @Override
+    protected synchronized void startScan() {
+        logger.trace("startScan() called.");
+
+        logger.debug("startScan(): creating a thing of type binding.");
+        ThingUID thingUID = new ThingUID(THING_TYPE_BINDING, "org_openhab_binding_velux");
+        // @formatter:off
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID)
+                    .withProperty(VeluxBindingProperties.PROPERTY_BINDING_BUNDLEVERSION, ManifestInformation.getBundleVersion())
+                    // TODO: should work with:
```

Comment:
If there is a solution for the i18 translation issue, I would really **appreciate** a pointer!
Unfortunatelly
` .withLabel("@text/binding.velux.name")`
does not lead to the defined text, neither to standard language nor to localization.

Reply:
I vaguely remember @lolodomo & @cweitkamp working with this, can any of you give a pointer here?

Reply:
This feature seem to be missing in our docs. I will add it. In the meantime you can find an example in the DarkSky binding:

1. Use any English label of your choice to create your `DiscoveryResult`:

https://github.com/openhab/openhab2-addons/blob/50d850428573c3230c3835131868a5ad209e91bb/bundles/org.openhab.binding.darksky/src/main/java/org/openhab/binding/darksky/internal/discovery/DarkSkyDiscoveryService.java#L133

2. Add the same English label to a default `ESH-INF/i18n/XXX.properties` file (`<bridge-id>` is optional):

```
discovery.<binding-id>.<thing-type-id>.<bridge-id>.<thing-id>.label = <your-english-label>
```

https://github.com/openhab/openhab2-addons/blob/50d850428573c3230c3835131868a5ad209e91bb/bundles/org.openhab.binding.darksky/src/main/resources/ESH-INF/i18n/darksky.properties#L13-L14

3. Add as much translations as you like following the guidelines of 2 (e.g. `ESH-INF/i18n/XXX_de.properties`):

https://github.com/openhab/openhab2-addons/blob/50d850428573c3230c3835131868a5ad209e91bb/bundles/org.openhab.binding.darksky/src/main/resources/ESH-INF/i18n/darksky_de.properties#L388-L389

**NOTE:** To make it work you have to inject references to the `LocaleProvider` and the `TranslationProvider` services to your `DiscoveryService` implementation.

https://github.com/openhab/openhab2-addons/blob/50d850428573c3230c3835131868a5ad209e91bb/bundles/org.openhab.binding.darksky/src/main/java/org/openhab/binding/darksky/internal/discovery/DarkSkyDiscoveryService.java#L58-L64

Reply:
Thanks, Christoph @cweitkamp, for your response.

 Now, the discovery result is created with 
`	.withLabel("Velux Binding“)`
 combined with all the i18n and locale stuff.

 Within i18n/velux.properties there is now the appropriate entry:

> 	discovery.velux.binding...label = Velux Binding

 with the corresponding entries with
	`src/main/resources/ESH-INF/i18n/velux_de.properties`:		
> discovery.velux.binding...label = Velux Binding (de) 

	`src/main/resources/ESH-INF/i18n/velux_dk.properties`:
> discovery.velux.binding...label = Velux Binding (dk) 

	`src/main/resources/ESH-INF/i18n/velux_nl.properties`:
> discovery.velux.binding...label = Velux Binding (nl)

 But those names do not appear within paperUI. Any idea? Does the empty bridge/thing dientifier confuse?

Best regards, Guenther

Reply:
You need to inject the localisation services and set them on the variables in the parent discovery class. See also https://github.com/openhab/openhab-addons/blob/master/bundles/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/discovery/DSMRBridgeDiscoveryService.java But better use constructor injecting instead of the setter methods.

Reply:
Thanks. It seems to work.